controllers/machine_controller_noderef.go

```diff
@@ -85,6 +86,14 @@ func (r *MachineReconciler) reconcileNode(ctx context.Context, cluster *clusterv
 		r.recorder.Event(machine, corev1.EventTypeNormal, "SuccessfulSetNodeRef", machine.Status.NodeRef.Name)
 	}
 
+	// Reconcile node annotations.
+	annotations.SetNodeAnnotation(node, clusterv1.ClusterNameAnnotation, machine.Spec.ClusterName)
+	annotations.SetNodeAnnotation(node, clusterv1.ClusterNamespaceAnnotation, machine.GetNamespace())
+	annotations.SetNodeAnnotation(node, clusterv1.MachineAnnotation, machine.Name)
+	if err := remoteClient.Patch(ctx, node, client.Merge); err != nil {
+		log.V(2).Info("Failed patch node to set annotations", "err", err, "node name", node.Name)
+	}
```

Comment:
If we're not using our patch helper, we should probably check if there are any changes here before performing the patch, otherwise we'd end up patching at every reconciliation

Reply:
done